lib/internal/buffer.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+if (!process.binding('config').hasIntl) {
+  return;
+}
+
+const normalizeEncoding = require('internal/util').normalizeEncoding;
+const Buffer = require('buffer').Buffer;
+
+const icu = process.binding('icu');
+
+// Maps the supported transcoding conversions. The top key is the from_enc,
+// the child key is the to_enc. The value is the transcoding function to.
+const conversions = {
+  'ascii': {
+    'latin1': (source) => {
+      return Buffer.from(source);
+    },
+    'utf8': (source) => {
+      return Buffer.from(source);
+    },
+    'utf16le': (source) => {
+      return icu.convertToUcs2('us-ascii', source);
+    },
+    'base64': (source) => {
+      return source.toString('base64');
+    },
+    'hex': (source) => {
+      return source.toString('hex');
```

Comment:
Mhhh this returns a `string` or a `Buffer` depending on the target encoding? I don’t think binary-to-text encodings should be allowed here, `.toString()` is the right method for them.